composer/loggers/slack_logger.py

```diff
@@ -1,64 +1,192 @@
+        include_keys: Optional[Set[str]] = None,
+        interval_in_seconds: Optional[int] = None,
+    ) -> None:
+        self.formatter_func = formatter_func
+        self.include_keys = include_keys
+        self.interval_in_seconds = interval_in_seconds
 
 
 class SlackLogger(LoggerDestination):
-    """Log metrics to slack, using `Slack Webhook URL <https://api.slack.com/messaging/webhooks>`.
+    """Logger to log metrics to slack.
 
-    .. note::
-        See example instantiating of SlackLogger in `examples/slack_log_metrics.py`
-        Webhook URL will be passed as an environment variable.
-        Formatter function can be defined at callsite or default formatter function will be applied if not provided.
+    This logger uses Slack's postMessage api - https://api.slack.com/methods/chat.postMessage.
 
     Args:
-        webhook_url (str): Slack Webhook URL
-        log_metrics_formatter_func ((...) -> Any | None): A formatter function that returns list of blocks to be sent to slack
+        log_metrics_config (SlackLogConfig): Configuration for logging metrics to slack.
+        log_traces_config (SlackLogConfig): Configuration for logging traces to slack.
+        log_hparams_config (SlackLogConfig): Configuration for logging hyperparameters to slack.
     """
 
     def __init__(
         self,
-        webhook_url: str = '',
-        log_metrics_formatter_func: Optional[Callable] = None,
+        log_metrics_config: Optional[SlackLogConfig] = None,
+        log_traces_config: Optional[SlackLogConfig] = None,
+        log_hparams_config: Optional[SlackLogConfig] = None,
     ) -> None:
-        self.webhook_url = webhook_url
-        self.logged_metrics: Dict[str, float] = {}
-        self.log_metrics_formatter_func = log_metrics_formatter_func if log_metrics_formatter_func else self._default_log_bold_key_normal_value_pair_with_header
+        try:
+            import slack_sdk
+            del slack_sdk
+        except ImportError as e:
+            raise MissingConditionalImportError('health_checker', 'slack_sdk', None) from e
+
+        self.client = WebClient()
+        self.slack_logging_api_key = os.environ.get('SLACK_LOGGING_API_KEY', None)
+        self.channel_id = os.environ.get('SLACK_LOGGING_CHANNEL_ID', None)
+
+        if self.slack_logging_api_key is None or self.channel_id is None:
+            raise RuntimeError(
+                'SLACK_LOGGING_API_KEY and SLACK_LOGGING_CHANNEL_ID must be set as environment variables')
+
+        self.log_metrics_config = log_metrics_config
+        self.log_traces_config = log_traces_config
+        self.log_hparams_config = log_hparams_config
+
+        # Maintain logger state
+        now = time.time()
+        self.logs_since_last_log, self.last_log_time = list(Any), now
+        self.traces_since_last_log, self.last_trace_time = list(Any), now
+        self.hparams_since_last_log, self.last_hparams_time = list(Any), now
```

Comment:
Allows different time intervals and formats for sending logs / traces / hparams. In most cases, user would implement just one out of these 3